lms/static/scripts/frontend_apps/components/SubmitGradeForm.js

```diff
@@ -20,18 +20,46 @@ import ValidationMessage from './ValidationMessage';
 export default function SubmitGradeForm({ disabled = false, student }) {
   const [showError, setShowError] = useState(false); // Is there a validation error message to show?
   const [gradeErrorMessage, setGradeErrorMessage] = useState(''); // The actual validation error message.
+  const [grade, setGrade] = useState('');
 
-  const [networkError, setNetworkError] = useState(''); // if there is an error when submitting grade
-  const [requestStatus, setRequestStatus] = useState(''); // ajax request state, one of ('', 'fetching', 'error', 'done')
+  const [networkError, setNetworkError] = useState(''); // If there is an error when submitting grade
+  const [requestStatus, setRequestStatus] = useState(''); // Ajax request state, one of ('', 'fetching', 'error', 'done')
+  const [fetchStatus, setFetchStatus] = useState(''); // Ajax request state, one of ('', 'fetching', 'error', 'done')
 
   const { authToken } = useContext(Config);
 
+  // Used to handle keyboard input effects.
+  const refInput = useRef(null);
+
   useEffect(() => {
     setRequestStatus(''); // clear out network status
-  }, [student]);
 
-  // Used to handle keyboard input effects.
-  const refInput = useRef(null);
+    // Fetch the grade from the service api
+    let fetchedGradeDone = false; // flag to prevent updating more than once
```

Comment:
Can you help me understand how "updating more than once" happens?

Reply:
This is some check to prevent multiple state updates if `useEffect` gets triggered twice before the network request finished. I will work on cleaning this up. Its also possible we don't really need this robust logic because it may be a behind side effect in the first place. I'll look into this.

Reply:
I haven't looked over this file closely, but one thing I note is that the flag is created inside the closure, which means that each time this component is rendered and `useEffect` is called, the flag will be a different variable. This means the `if (!fetchedGradeDone)` check lower done will always evaluate to `true`.